src/query/api/v1/handler/influxdb/write.go

```diff
@@ -0,0 +1,255 @@
+/*
+ * Author: Markus Stenberg <mstenber@aiven.io>
+ *
+ * Copyright (c) 2019 Aiven Oy
+ */
+
+package influxdb
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	imodels "github.com/influxdata/influxdb/models"
+	"github.com/m3db/m3/src/cmd/services/m3coordinator/ingest"
+	"github.com/m3db/m3/src/dbnode/client"
+	"github.com/m3db/m3/src/query/api/v1/handler"
+	"github.com/m3db/m3/src/query/models"
+	"github.com/m3db/m3/src/query/ts"
+	"github.com/m3db/m3/src/query/util/logging"
+	xerrors "github.com/m3db/m3/src/x/errors"
+	"github.com/m3db/m3/src/x/instrument"
+	xhttp "github.com/m3db/m3/src/x/net/http"
+	xtime "github.com/m3db/m3/src/x/time"
+	"go.uber.org/zap"
+)
+
+const (
+	InfluxWriteURL = handler.RoutePrefixV1 + "/influxdb/write"
+)
+
+type ingestWriteHandler struct {
+	downsamplerAndWriter ingest.DownsamplerAndWriter
+	instrumentOpts       instrument.Options
+	promRewriter         *promRewriter
+}
+
+type ingestField struct {
+	name  []byte // to be stored in __name__; rest of tags stay constant for the Point
+	value float64
+}
+
+type ingestIterator struct {
+	// what is being iterated
+	points       []imodels.Point
+	promRewriter *promRewriter
+	pointIndex   int
+	err          xerrors.MultiError
+
+	// following entries are within current point, and initialized
+	// when we go to the first entry in the current point
+	fields         []*ingestField
+	nextFieldIndex int
+	tags           models.Tags
+}
+
+func (self *ingestIterator) populateFields() bool {
+	point := self.points[self.pointIndex]
+	it := point.FieldIterator()
+	n := 0
+	for it.Next() {
+		n += 1
+	}
+	if n == 0 {
+		return false
+	}
+	it.Reset()
+	self.fields = make([]*ingestField, 0, n)
+	for it.Next() {
+		var value float64 = 0
+		switch it.Type() {
+		case imodels.Boolean:
+			v, err := it.BooleanValue()
+			if err != nil {
+				err = fmt.Errorf("Error decoding boolean: %w", err)
+				self.err = self.err.Add(err)
+				continue
+			}
+			if v {
+				value = 1.0
+			}
+		case imodels.Integer:
+			v, err := it.IntegerValue()
+			if err != nil {
+				err = fmt.Errorf("Error decoding integer: %w", err)
```

Comment:
nit: errors should start with lower case; also looks like the [influx errors](https://github.com/influxdata/influxdb/blob/138caab0a2a9175763be47e6b37d8ddc53eb4ae4/models/points.go#L2479) are quite verbose already, so you should be able to just use `err` without wrapping it further